src/kube-runtime/src/init.d/user_command_renderer.py

```diff
@@ -0,0 +1,90 @@
+#!/usr/bin/python
+
+# Copyright (c) Microsoft Corporation
+# All rights reserved.
+#
+# MIT License
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy of this software and associated
+# documentation files (the "Software"), to deal in the Software without restriction, including without limitation
+# the rights to use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the Software, and
+# to permit persons to whom the Software is furnished to do so, subject to the following conditions:
+# The above copyright notice and this permission notice shall be included in all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED *AS IS*, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING
+# BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+# NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM,
+# DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+
+import argparse
+import logging
+import os
+import re
+import sys
+
+import pystache
+import yaml
+
+sys.path.append(os.path.join(os.path.dirname(os.path.abspath(__file__)), ".."))
+from common.utils import init_logger  #pylint: disable=wrong-import-position
+
+LOGGER = logging.getLogger(__name__)
+
+
+def _convert_to_dict(obj) -> dict:
+    converted_obj = {}
+    if isinstance(obj, list):
+        for i, value in enumerate(obj):
+            converted_obj[str(i)] = value
+    elif isinstance(obj, dict):
+        for key, value in obj.items():
+            converted_obj[key] = _convert_to_dict(value)
+    else:
+        converted_obj = obj
+    return converted_obj
```

Comment:
what is this function for

Reply:
According to issue: https://github.com/defunkt/pystache/issues/196
pystache not support syntax such as `array.0`, we need to convert array to dict first, then we can access array item.